scripts/slack-music-notifier.js

```diff
@@ -1,56 +1,106 @@
-(function() {
+(function(window, $, storage) {
   'use strict';
-  var title;
+  var SPOTIFY = "spotify";
 
-  function checkSong(service, options) {
+  function checkSong(service, options, prevSong) {
+    var title;   
+    prevSong = prevSong || {};
+    
     // Delay posting song for 15 seconds to not report skipped songs
     var currentPlayTime = $(options[service].playtime).text();
     var playTimeSecs = hmsToSecondsOnly(currentPlayTime);
-    if (playTimeSecs < 15) {
+    if (!playTimeSecs || playTimeSecs < 15) {
       setTimeout(function() {
-        checkSong(service, options);
+        checkSong(service, options, prevSong);
       }, 1000);
       return;
-    }
+    }    
 
     // Pandora fades album covers, wait until the new cover is loaded
-    var opacity = $(options[service].cover).css('opacity');
-    if (opacity != 1) {
+    if (options[service].cover) {
+      var opacity = $(options[service].cover).css('opacity');
+      if (opacity != 1) {
+        setTimeout(function() {
+          checkSong(service, options, prevSong);
+        }, 1000);
+        return;
+      }
+    }
+  
+    // Check to see if this is the same as the previous song
+    title = $(options[service].title).text();
+    if (prevSong.title == title) {
       setTimeout(function() {
-        checkSong(service, options);
+          checkSong(service, options, prevSong);
       }, 1000);
       return;
     }
-
-    var song = {
-      artist: $(options[service].artist).text(),
-      album: $(options[service].album).text(),
-      title: $(options[service].title).text(),
-      cover: $(options[service].cover).prop('src')
-    }
-
-    chrome.storage.local.get( service + '-song', function(stored) {
-      if (!isEmptySong(song) && 
-          (!stored[service + '-song'] || !isSameSong(stored[service + '-song'], song))) {
+  
+    // Spotify doesn't provide album information in their Now Playing
+    if (service === SPOTIFY) {
+      var element = $(options.spotify.elementSelector);
+      var uri = options.spotify.url;
+      var id = element[0].dataset.uri;
+    
+      if (!id) {
+        setTimeout(function() {
+          checkSong(service, options, prevSong);
+        }, 1000);
+        return;
+      }
+    
+      uri += id.replace(options.spotify.regExp, '');
+      $.get(uri, function(data) {
+        var song = {
+          artist: data.artists[0].name,
+          album: data.album.name,
+          title: title,
```

Comment:
Probably use the title returned by the API to normalize it? Are they always the same?